static/test/javascripts/spec/commercial/hosted/next-video-autoplay.spec.js

```diff
@@ -0,0 +1,94 @@
+
+        var fixturesConfig = {
+                id: 'next-video-autoplay',
+                fixtures: [
+                    '<div><video data-duration="160">' +
+                    '<source type="video/mp4" src="">' +
+                    '</video></div>' +
+                    '<div class="js-hosted-next-autoplay">' +
+                    '<div class="js-autoplay-timer" data-next-page="/commercial/advertiser-content/renault-car-of-the-future/design-competition-episode2">10s</div>' +
+                    '</div>' +
+                    '<button class="js-autoplay-cancel">' +
+                    '</button>'
+                ]
+            },
+            $fixturesContainer;
+
+        beforeEach(function (done) {
+            injector.mock('common/modules/analytics/google', function noop() {});
+            injector.mock('commercial/modules/hosted/next-video',  {
+                load: Promise.resolve
+            });
+            injector.require([
+                'commercial/modules/hosted/next-video-autoplay'
+            ], function () {
+                nextVideoAutoplay = arguments[0];
+
+                $fixturesContainer = fixtures.render(fixturesConfig);
+                nextVideoAutoplay.init().then(done);
+            });
+        });
+
+        afterEach(function () {
+            fixtures.clean(fixturesConfig.id);
+        });
+
+        it('should exist', function (done) {
+            expect(nextVideoAutoplay).toBeDefined();
+            done();
+        });
+
+        it('should trigger autoplay when there is a next video', function (done) {
+            expect(nextVideoAutoplay.canAutoplay()).toBeTruthy();
+            done();
+        });
+
+        it('should show end slate information', function (done) {
+            nextVideoAutoplay.triggerEndSlate();
+            fastdom.defer(1, function () {
+                expect($('.js-hosted-next-autoplay', $fixturesContainer).hasClass('js-autoplay-start')).toBeTruthy();
+                done();
+            });
+        });
+
+        it('should hide end slate information when cancel button is clicked', function (done) {
+            nextVideoAutoplay.addCancelListener();
+            document.querySelector('.js-autoplay-cancel').click();
+            fastdom.defer(1, function () {
+                expect($('.js-hosted-next-autoplay', $fixturesContainer).hasClass('hosted-slide-out')).toBeTruthy();
+                done();
+            });
+        });
+
+        it('should not trigger autoplay when there is no next video', function (done) {
+            fixturesConfig = {
+                id: 'next-video-autoplay',
+                fixtures: [
+                    '<div class="js-autoplay-timer" data-next-page="">10s</div>'
+                ]
+            };
+            $fixturesContainer = fixtures.render(fixturesConfig);
+            nextVideoAutoplay.init().then(function(){
+                expect(nextVideoAutoplay.canAutoplay()).toBeFalsy();
+                done();
```

Comment:
Well, come to think of it. You will create two `DIV` with the same `id`. I think this will have side-effects.